rellic/AST/Z3ConvVisitor.cpp

```diff
@@ -869,38 +896,37 @@ void Z3ConvVisitor::VisitBinaryApp(z3::expr z_op) {
   clang::Expr *c_op{nullptr};
   switch (z_func.decl_kind()) {
     case Z3_OP_EQ:
-      c_op = CreateBinaryOperator(*ast_ctx, clang::BO_EQ, lhs, rhs,
-                                  ast_ctx->IntTy);
+      c_op = BinOpExpr(clang::BO_EQ, ast_ctx->IntTy);
       break;
 
     case Z3_OP_SLEQ:
-      c_op = CreateBinaryOperator(*ast_ctx, clang::BO_LE, lhs, rhs,
-                                  ast_ctx->IntTy);
+      c_op = BinOpExpr(clang::BO_LE, ast_ctx->IntTy);
       break;
 
     case Z3_OP_AND: {
```

Comment:
Can you add a comment here that says something like `AND`s in z3 can be variadic, and so you lower / fold them into something like `(A & (B & C))` or `((A & B) & C)`, and make a note of `lhs` and `rhs` being shared byref with `BinOpExpr`.

Reply:
Are there any possible downstream interactions with the order of the `&&` or `||` folding? I.e. if it's `(A && (B && ...))` vs. `((...) && C)`, will the short-circuiting of `C` interact in any way? I guess it depends on if a result of an effectful result is stored to something like `C`, and making sure it's not inlined in place of `C`.